safenode/src/network/msg/mod.rs

```diff
@@ -44,7 +44,7 @@ impl SwarmDriver {
                     let _ = self
                         .pending_requests
                         .remove(&request_id)
-                        .ok_or(Error::Other("Request to still be pending".to_string()))?
+                        .ok_or(Error::ReceivedResponseDropped(request_id))?
                         .send(Ok(response));
```

Comment:
The result after sending the response is ignored here, maybe it can be mapped to `.map_err(|_| Error::InternalMsgChannelDropped)?`